modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/set_permissions_command.rb

```diff
@@ -0,0 +1,156 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Peripherals
+    module StorageInteraction
+      module OneDrive
+        class SetPermissionsCommand
+          using ServiceResultRefinements
+
+          def self.call(storage:, path:, permissions:)
+            new(storage).call(path:, permissions:)
+          end
+
+          def initialize(storage)
+            @storage = storage
+          end
+
+          def call(path:, permissions:)
+            # This could be broken down into:
+            # 1. Get Permission IDs
+            # 2. Apply Permission Changes
+            # 2.1 Create Permissions
+            # 2.2 Delete Permissions
+            # 2.3 Re-create Permissions
```

Comment:
🟡 is that true? from code perspective, I cannot see a point `2.1 Create Permissions`, see line 85

Reply:
When there are no permissions set, we need to create them. In other words, with a new folder we can't delete permissions because they don't exist and need to create new.

Does this answer your questions or am I being extra dense (probably the case)?

Reply:
probably just a misunderstanding of the words of the comment. Looking at the code, I'd write the comment as:

```
# 2. Apply Permission Changes
# 2.1 Delete Permissions (skipped if not existent)
# 2.2 (Re-)Create Permissions
```